pipeline/types/engine.py

```diff
@@ -54,3 +54,10 @@ class MatchingOutput:
     users: List[User]
     matches: List[Match]
     internal_data: MatchingInternalData = MatchingInternalData()
+
+
+@dataclass
+class MatchingInternalAnalytics:  # More metrics will be added in the future
+    n_proposed_matches_per_user: Dict = field(
+        default_factory=dict
+    )  # A dictionary where keys -> users and values -> metrics
```

Comment:
As a coding style point, I think it is easier to read and understand if the comments are on their own line, above the section they refer to. Like this:

```python
@dataclass
class MatchingInternalAnalytics: 
    # A dictionary where keys -> users and values -> metrics
    n_proposed_matches_per_user: Dict = field(
        default_factory=dict
    ) 
    # More metrics will be added in the future
```